sdk/rust/src/http/router.rs

```diff
@@ -103,79 +105,96 @@ impl Router {
     }
 
     /// Register a handler at the path for all methods.
-    pub fn any<F>(&mut self, path: &str, handler: F)
+    pub fn any<F, I, O>(&mut self, path: &str, handler: F)
     where
-        F: Fn(Request, Params) -> Result<Response> + 'static,
+        F: Fn(I, Params) -> O + 'static,
+        I: From<Request>,
+        O: IntoResponse,
     {
-        self.any_methods.add(path, Box::new(handler)).unwrap();
+        self.any_methods
+            .add(
+                path,
+                Box::new(move |req, params| handler(req.into(), params).into_response()),
+            )
+            .unwrap();
     }
 
     /// Register a handler at the path for the specified HTTP method.
-    pub fn add<F>(&mut self, path: &str, method: http_types::Method, handler: F)
+    pub fn add<F, I, O>(&mut self, path: &str, method: Method, handler: F)
     where
-        F: Fn(Request, Params) -> Result<Response> + 'static,
+        F: Fn(I, Params) -> O + 'static,
+        I: TryFrom<Request>,
```

Comment:
I think we probably want to make a `FromRequest` trait. I'll be honest, that at this point I'm slowly moving toward 100% custom conversion traits not because we need them right now, but because the orphan rule scares me.